### ISSUE TITLE
[mu4e bug] `mu find…` doesn't reflect changes done in mu4e if changes are solely deletions

### ISSUE THREAD

**Issue by mbunkus** [djcb/mu#2302]
This is similar to #2100.

If you have new & unread emails, and you delete them all via `mu4e-headers-mark-for-delete` & subsequent apply, then an external run of `mu find…` will still find those new & unread emails as new & unread, even though they've been deleted in `mu4e`.

This does not happen if any other change is done at the same time, e.g. if you mark one mail for deletion and another for trashing. In that case the changes are reflected in `mu find…` correctly as neither of the two continues to show up as new & unread.

**How to Reproduce**

1. Send yourself two emails & let `mu4e` fetch them.
2. Make sure you can see both new & unread mails in `mu4e` and via something like `mu find flag:unread`.
3. Mark both emails for deletion & apply the deletion.
4. Run `mu find flag:unread` again & observe that the mails are still found.
5. Wait a while & repeat step 4; they're still found.

Now for the other case:

1. Send yourself two emails & let `mu4e` fetch them.
2. Make sure you can see both new & unread mails in `mu4e` and via something like `mu find flag:unread`.
3. Mark one for deletion & the other trashing & apply both.
4. Run `mu find flag:unread` again & observe that the mails are not found anymore.

**Environment**

* Emacs 28.1
* mu 1.8.6
* ArchLinux, both from Arch's packages

**Checklist**
- [x] you are running either the latest 1.6.x release, or a 1.8.x release (otherwise, please upgrade)
- [x] you are running mu4e without any third-party extensions (otherwise, make sure you can reproduce without those)
- [x] you have read all of the above

**Comment by djcb** [comment]
The only way to have the mu4e (in memory) database immediately available when accessing it external (i.e. through mu fine) would be to `commit` after each command. That slows down mu4e noticeably (I tried). So, you're always going to have cases where the command-line is a bit behind (perhaps we could commit when idle, or every _n_ seconds..., not sure if it's worth the extra complexity).

Now, the report _does_ show a bug though, `mu find` should not include non-existing messages in its results. That I have fixed, and will be in the next 1.8 release.
